common/djangoapps/course_modes/views.py

```diff
@@ -38,6 +38,10 @@ class ChooseModeView(View):
 
     """
 
+    @method_decorator(transaction.non_atomic_requests)
+    def dispatch(self, *args, **kwargs):
+        return super(ChooseModeView, self).dispatch(*args, **kwargs)
+
     @method_decorator(login_required)
```

Comment:
FWIW, I was assuming that a `get()` wouldn't perform any DB changes - but I shouldn't assume that.